### PR TITLE
feat: add support for custom principal iam arns

### PR DESCRIPTION
**Pull request by SushiFu**

### CODE REVIEW

**Review by npalm — COMMENTED** [review]
Thanks for your contribution. Will run a quick test in the next days

**Review by npalm — CHANGES_REQUESTED** [review]
Looks good, a small remark. I think we can directly use the array in the policy. @GuptaNavdeep1983 I think this is a better approach as the hard coded root principal we have. What do you think?

File: main.tf [hunk 1]
```diff
@@ -59,6 +59,17 @@ data "aws_iam_policy_document" "github_actions_assume_role_policy" {
...
     }
   }
+  dynamic "statement" {
+    for_each = toset(var.custom_principal_arns)
+    content {
+      actions = ["sts:AssumeRole"]
+
+      principals {
+        type        = "AWS"
+        identifiers = [statement.value]
```
[the remark this answers is not in the document]
Comment (SushiFu) [reply main.tf]: Agreed I have refactored to have only one statement for AWS principals

**Review by GuptaNavdeep1983 — APPROVED** [review]

**Review by npalm — COMMENTED** [review]
Thanks, change looks good. Need to run a quick deployment test

**Review by npalm — APPROVED** [review]

**Pull request merged**